core/container/dockercontroller/dockercontroller.go

```diff
@@ -26,16 +26,32 @@ import (
 	"github.com/hyperledger/fabric/core/container/ccintf"
 	cutil "github.com/hyperledger/fabric/core/container/util"
 	"github.com/op/go-logging"
+	"github.com/spf13/viper"
 	"golang.org/x/net/context"
 )
 
-var dockerLogger = logging.MustGetLogger("dockercontroller")
+var (
+	dockerLogger = logging.MustGetLogger("dockercontroller")
+	hostConfig   = new(docker.HostConfig)
+)
 
 //DockerVM is a vm. It is identified by an image id
 type DockerVM struct {
 	id string
 }
 
+func init() {
+	err := viper.UnmarshalKey("vm.docker.hostConfig", hostConfig)
+	if err != nil {
+		dockerLogger.Fatalf("Load docker HostConfig wrong, error: %s", err.Error())
+	}
+
+	if hostConfig.NetworkMode == "" {
+		hostConfig.NetworkMode = "host"
+	}
+	dockerLogger.Debug("Load docker HostConfig: %+v", hostConfig)
+}
+
```

Comment:
Yes like that but combined with this work to read into a HostConfig in init(). For example something like
```
func init() {
	networkMode := viper.GetString("vm.docker.networkMode")
 	if networkMode == "" {
 		networkMode = "host"
 	}
    dns := viper.GetStringSlice("vm.docker.dns")        //use only supported properties
    hostConfig = docker.HostConfig{}
    hostConfig.NetworkMode = networkMode
    hostConfig.Dns = dns

    dockerLogger.Debug("Load docker HostConfig: %+v", hostConfig)
}
```

Will note that in my first comment, I was going for wrapping the docker.Client in another struct so we can embed the HostConfig in the wrapped struct.

But even referring to a global HostConfig constructed in init() as above is a step in the right direction.